test/rabintest.js

```diff
@@ -0,0 +1,173 @@
+let expect = require("chai").expect;
+const{ generateRabinKeyFromSeed,
+    createRabinSignature,
+    verifyRabinSignature } = require("../src/rabin");
+const { getRandomInt, getRandomHex } = require('../src/utils');
+
+let defaultSeed = '0x01';
+let defaultKey = generateRabinKeyFromSeed(defaultSeed);
+let defaultWrongKey = generateRabinKeyFromSeed(defaultSeed+'00');
+let defaultDataHex = Buffer.from("msg").toString('hex');
+let defaultIncorrectHexResultValue = "0xdeadbeef";
+let defaultIncorrectInputHexValue = "0x01";
+let defaultNotHexValue = "defaultNotHexValue";
+let defaultSignatureResult = createRabinSignature(defaultDataHex, defaultKey.p, defaultKey.q, defaultKey.nRabin);
+
+let seed2 = 'e8d999619a31a000187075157c2213001893c551e44532003cf52959aa1154';
+let key2 = generateRabinKeyFromSeed(seed2);
+
+describe("Key Generation Tests", function() {
+    describe("Correct Result Tests", function() {
+        it("Sample Seed 1", function() {
+          let key = generateRabinKeyFromSeed(defaultSeed);
+          expect(key.p).to.equal("0x1a022b166eeb68c189739189af9718d2b302f08836c8a6dc2ba9de3badb1ef5c712e1ec1eeabd173407bb12ddc1e9fe01f18e96a01f74cbbad94d7a1ed7597");
```

Comment:
These hardcoded numbers do not test the code well since they are generated from the code itself, right? Let's just do 

1. generate keys
2. sign
3. verify: it use results from step 1 & 2 -> good; otherwise -> bad

We can do it multiple rounds by using different keys and signing messages.

Reply:
I do this x10 as the final test

Reply:
I saw it, very nice.